framework/wazuh/agent.py

```diff
@@ -607,7 +607,7 @@ def get_agent_groups(group_list: list = None, offset: int = 0, limit: int = None
 
         rbac_filters = get_rbac_filters(system_resources=system_groups, permitted_resources=group_list)
 
-        with WazuhDBQueryGroup(**rbac_filters) as group_query:
+        with WazuhDBQueryGroup(**rbac_filters, limit=limit) as group_query:
```

Comment:
We need to send the offset parameter to the DB so it fetches the expected data:
```suggestion
        with WazuhDBQueryGroup(**rbac_filters, limit=limit, offset=offset) as group_query:
```

Also, remove `limit` and `offset` from process_array on line 630:
https://github.com/wazuh/wazuh/blob/958f6e2264d7e3dd5fdad67d3d73c24cf0743623/framework/wazuh/agent.py#L630